src/lib/core/datetime.c

```diff
@@ -331,27 +331,15 @@ datetime_parse_full(struct datetime *date, const char *str, size_t len,
 	if (len <= 0)
 		goto exit;
 
-	/* now we have parsed enough of date literal, and we are
-	 * ready to consume timezone suffix, if overridden
-	 */
-	time_t base = dt_epoch(dt) + sec_of_day - offset * 60;
-	ssize_t l;
-	if (tzsuffix != NULL) {
-		l = parse_tz_suffix(tzsuffix, strlen(tzsuffix), base,
-				    &tzindex, &offset);
-		if (l < 0)
-			return l;
-		goto exit;
-	}
-
 	if (*str == ' ') {
 		str++;
 		len--;
 	}
 	if (len <= 0)
 		goto exit;
 
-	l = parse_tz_suffix(str, len, base, &tzindex, &offset);
+	time_t base = dt_epoch(dt) + sec_of_day - offset * 60;
```

Comment:
Should you use `offset` here? I think it's always `0`.

Reply:
Actually, this patch is a part of https://github.com/tarantool/tarantool/pull/8412.

Removed:

```diff
+++ b/src/lib/core/datetime.c
@@ -338,7 +338,7 @@ datetime_parse_full(struct datetime *date, const char *str, size_t len)
        if (len <= 0)
                goto exit;
 
-       time_t base = dt_epoch(dt) + sec_of_day - offset * 60;
+       time_t base = dt_epoch(dt) + sec_of_day;
        ssize_t l = parse_tz_suffix(str, len, base, &tzindex, &offset);
        if (l < 0)
                return l;
```

Branch `tsafin/datetime-format-tzname` has been force-pushed.